File: app/schemas/recibo.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import date

class ImputacionCreate(BaseModel):
    comprobante_deuda_id: int
    importe: float

class ReciboCreate(BaseModel):
    cliente_id: int
    fecha_emision: date
    punto_venta: int
    total: float
    observaciones: Optional[str] = None
    imputaciones: List[ImputacionCreate] = []

class ReciboResponse(BaseModel):
    id: int
    numero: int
    fecha_emision: date
    total: float
    saldo: float
    cliente_id: int
    nombre_cliente: Optional[str] = None
    observaciones: Optional[str]
    # Podríamos agregar detalles de imputaciones aquí si fuera necesario
    
    class Config:
        from_attributes = True

File: frontend/src/modules/Tesoreria/domain/models/Recibo.ts
export interface ImputacionCreate {
    comprobante_deuda_id: number;
    importe: number;
}

export interface ReciboCreate {
    cliente_id: number;
    fecha_emision: string; // YYYY-MM-DD
    punto_venta: number;
    total: number;
    observaciones?: string;
    imputaciones: ImputacionCreate[];
}

export interface ReciboResponse {
    id: number;
    numero: number;
    fecha_emision: string;
    total: number;
    saldo: number;
    cliente_id: number;
    nombre_cliente?: string;
    observaciones?: string;
}
